python/bolt-mcp/src/bolt_mcp/oauth/endpoints.py

```
@@ -0,0 +1,290 @@
+        grant_type = form.get("grant_type")
+        if grant_type == "authorization_code":
+            return await _token_authorization_code(form)
+        if grant_type == "refresh_token":
+            return await _token_refresh(form)
+        return _json_error("unsupported_grant_type", f"unsupported grant_type: {grant_type!r}")
+
+    # ── POST {prefix}/register (RFC 7591 DCR) ────────────────────────────────────
+    async def _register(request: Request):
+        try:
+            body = json_decode(request.body or b"{}")
+        except Exception:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)